membership_extension/models/membership_line.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import timedelta
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning as UserError
+from openerp.addons.membership.membership import STATE
+
+
+class MembershipLine(models.Model):
+    _inherit = "membership.membership_line"
+    _order = 'date_to desc, id desc'
+
+    category_id = fields.Many2one(
+        string="Category", comodel_name='membership.membership_category',
+        related="membership_id.membership_category_id", readonly=True)
+    date_from = fields.Date(readonly=False)
+    date_to = fields.Date(readonly=False)
+    state = fields.Selection(
+        selection=STATE, string="Membership Status", readonly=False,
+        compute='_compute_state', inverse="_inverse_state",
+        help="""It indicates the membership status.
+        - Non Member: A member who has not applied for any membership.
+        - Cancelled Member: A member who has cancelled his membership.
+        - Old Member: A member whose membership date has expired.
+        - Waiting Member: A member who has applied for the membership and
+                          whose invoice is going to be created.
+        - Invoiced Member: A member whose invoice has been created.
+        - Paid Member: A member who has paid the membership amount."""
+    )
+
+    def __init__(self, pool, cr):
+        super(MembershipLine, self).__init__(pool, cr)
+        for model, store in pool._store_function.iteritems():
+            pool._store_function[model] = [
+                x for x in store
+                if x[0] != 'membership.membership_line' and x[1] != 'state']
+
+    @api.multi
+    def theorical_dates_set(self):
+        for line in self:
+            if line.date and line.membership_id:
+                date = fields.Date.from_string(line.date)
+                line.date_from = line.date
+                next_date = line.membership_id._get_next_date(date)
+                if next_date:
+                    date_to = fields.Date.to_string(next_date - timedelta(1))
+                    if date_to >= line.date:
+                        line.date_to = date_to
+
+    @api.onchange('membership_id')
+    def _onchange_membership_id(self):
+        self.member_price = self.membership_id.lst_price
+        self.theorical_dates_set()
+
+    @api.onchange('date')
+    def _onchange_date(self):
+        self.theorical_dates_set()
+
+    @api.multi
+    def _compute_state(self):
```

Comment:
Please put a comment about this nasty trick then (you are just omitting the write here in the method as I see, isn't it?). Isn't the field reset when entering the method?